engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/ModifyProcessInstanceTest.java

```diff
@@ -672,6 +675,67 @@ public void shouldSetVariablesFromMultipleInstructions() {
             tuple("bar", "true"));
   }
 
+  @Test
+  public void shouldTerminateAndActivateElementInTheSameScope() {
+    // given
+    ENGINE
+        .deployment()
+        .withXmlResource(
+            Bpmn.createExecutableProcess(PROCESS_ID)
+                .startEvent()
+                .subProcess(
+                    "sp",
+                    sp ->
+                        sp.embeddedSubProcess().startEvent().userTask("A").userTask("B").endEvent())
+                .endEvent()
+                .done())
+        .deploy();
+    final var processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    final var terminateElement =
+        RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .withElementId("A")
+            .getFirst();
+
+    // when
+    ENGINE
+        .processInstance()
+        .withInstanceKey(processInstanceKey)
+        .modification()
+        .terminateElement(terminateElement.getKey())
+        .activateElement("B")
+        .modify();
+
+    // then
+    Assertions.assertThat(
+            RecordingExporter.processInstanceRecords()
+                .onlyEvents()
+                .withProcessInstanceKey(processInstanceKey)
+                .withElementId("A")
+                .limit("A", ProcessInstanceIntent.ELEMENT_TERMINATED)
+                .toList())
+        .extracting(Record::getIntent)
+        .containsSequence(
+            ProcessInstanceIntent.ELEMENT_TERMINATING, ProcessInstanceIntent.ELEMENT_TERMINATED);
+    verifyThatElementIsActivated(
+        processInstanceKey,
+        "B",
+        BpmnElementType.USER_TASK,
+        terminateElement.getValue().getFlowScopeKey());
```

Comment:
This line (L725) makes sure that the activated element B is using the same flow scope as the terminated element instance, but it's a bit hidden.

🔧 It would be great if we could find a way to highlight this part of the verification, as it is essential to the implemented issue. Perhaps we should just add an additional explicit verification with a clear description.